runtime/manifest.js

```diff
@@ -143,8 +144,17 @@ class Manifest {
     let tags = options.tags || [];
     let subtype = options.subtype || false;
     function typePredicate(view) {
+      let resolvedType = type;
+      let primitiveType = type.isSetView ? type.primitiveType() : type;
```

Comment:
maybe fix & use `type.resolvedType()` to rebuild a type that has any contained type variables replaced with the resolved types. so `[~a].resolvedType` = `[Thing]`

Reply:
done